core/src/verify.rs

```diff
@@ -346,6 +368,17 @@ impl CommitSequenceVerifier {
                         Error::CryptoError("invalid agenda proof: invalid signature".to_string(), e)
                     })?;
                 }
+                // Verify agenda's last previous_block_hash matches with the real previous block hash
+                if CommitSequenceVerifier::verify_agenda_integrity(
+                    agenda,
+                    &self.header.to_hash256(),
+                )
+                .is_err()
+                {
+                    return Err(Error::InvalidArgument(
+                        "invalid agenda previous_block_hash".to_string(),
+                    ));
+                }
```

Comment:
Incoming commit is agenda proof commit. This code should be deleted.

Reply:
reflected